Update libunwind to v1.5.0

ararslan opened the pull request:
We were a few versions behind (previously 1.3.2). I've updated our patches accordingly with the upstream PRs for these changes against libunwind master. Also note that this is nongnu libunwind, not LLVM libunwind, so this only affects Linux and FreeBSD.

ararslan committed:
Update libunwind to v1.5.0

nongnu libunwind, not LLVM libunwind, so this only affects Linux and
FreeBSD

DilumAluthge commented:
@ararslan Can you rebase this on the latest master?

Also, it looks like you'll need to fix some merge conflicts.

vtjnash commented on deps/unwind.mk:
```diff
@@ -4,8 +4,15 @@ ifneq ($(USE_BINARYBUILDER_LIBUNWIND),1)
 LIBUNWIND_CFLAGS := -U_FORTIFY_SOURCE $(fPIC)
 LIBUNWIND_CPPFLAGS :=
 
+# XXX: Kludge for libunwind 1.5.0 tag name not being the full version
```
Could we open an issue there and ask them to push `v1.5.0` as a tag to github?

ararslan replied:
https://github.com/libunwind/libunwind/issues/309

giordano replied:
Fixed

vtjnash commented on deps/unwind.mk:
```diff
@@ -35,7 +42,7 @@ $(SRCCACHE)/libunwind-$(UNWIND_VER)/libunwind-dwarf-table.patch-applied: $(SRCCA
 	mkdir -p $(dir $@)
 	cd $(dir $@) && \
-	$(dir $<)/configure $(CONFIGURE_COMMON) CPPFLAGS="$(CPPFLAGS) $(LIBUNWIND_CPPFLAGS)" CFLAGS="$(CFLAGS) $(LIBUNWIND_CFLAGS)" --enable-shared --disable-minidebuginfo --disable-tests
+	$(dir $<)/configure $(CONFIGURE_COMMON) CPPFLAGS="$(CPPFLAGS) $(LIBUNWIND_CPPFLAGS)" CFLAGS="$(CFLAGS) $(LIBUNWIND_CFLAGS)" --enable-shared --disable-minidebuginfo --disable-tests --disable-zlibdebuginfo
```
We have zlib now included, so we should be able to include zlibdebuginfo

ararslan replied:
Good call, done

vtjnash reviewed (changes requested):
Missing update to stdlib/LibUnwind_jll/Project.toml file version

vtjnash commented:
Relevant PRs:
https://github.com/JuliaPackaging/Yggdrasil/pull/3804
https://github.com/JuliaPackaging/Yggdrasil/pull/3856
https://github.com/JuliaPackaging/Yggdrasil/pull/3859

Changes log:
https://github.com/libunwind/libunwind/compare/v1.3.2...v1.5

ararslan commented:
Weird failures on buildkite. Rebased on master, hopefully that fixes them.

giordano commented:
There is [v1.6.0](https://github.com/libunwind/libunwind/releases/tag/v1.6.0) now, you may want to try that one?

mkitti commented:
xref: https://github.com/ESMValGroup/ESMValTool/issues/2433 (libunwind 1.6 appears to breaking ABI compat with Julia built against prior versions via conda-forge)

vtjnash commented:
libunwind declares itself to be ABI-compatible with SONUMBER 8 for both, so you need to open an issue with them
```
$ objdump -x usr/lib/libunwind.so  | grep SONAME
  SONAME               libunwind.so.8
```

vtjnash commented:
defined here: https://github.com/libunwind/libunwind/blob/master/src/Makefile.am#L1

ararslan commented:
> There is v1.6.0 now ([...]), you may want to try that one?

Probably not given https://github.com/libunwind/libunwind/pull/295#issuecomment-981823223 though I guess that PR could be added as a patch.

All of the fixes and upstreaming PRs I've submitted to libunwind have been merged to master (though none are on 1.6-stable) so we should be able to drop a bunch of patches come libunwind 1.7.

ararslan commented:
And now there's a 1.6.1 :smile:

nalimilan commented:
Any news on this? FWIW I get a build failure on Fedora 36 when using libunwind 1.6.

ararslan commented:
> Any news on this?

It is news to me that anyone remembered or cared about this PR. :smile:

> FWIW I get a build failure on Fedora 36 when using libunwind 1.6.

Good to know. It seems that 1.5 works okay across the board so I'll leave it there and we can revisit other versions at some point.

I've rebased this branch on master and removed the tag version kludge as it's no longer necessary (upstream added a v1.5.0 alongside v1.5).

DilumAluthge commented:
The upstream is releasing new versions faster than Alex can update this PR 😂

DilumAluthge commented:
What's the plan for getting this PR merged?

ararslan commented:
> What's the plan for getting this PR merged?

@vtjnash approved back in November, so assuming CI passes with the rebase and Jameson is still on board then it should be good to merge I think.

ararslan commented:
CI failures:
- Linux x64 multithreaded: unrelated Distributed failure
   ```
   Test Failed at /cache/build/dockerized-amdci4-8/julialang/julia-master/julia-f2d3dce18c/share/julia/stdlib/v1.8/Distributed/test/distributed_exec.jl:301
     Expression: remotecall_fetch((k->begin
                   haskey(Distributed.PGRP.refs, k)
               end), wid1, rrid) == false
      Evaluated: true == false
   ERROR: LoadError: There was an error during testing
   in expression starting at /cache/build/dockerized-amdci4-8/julialang/julia-master/julia-f2d3dce18c/share/julia/stdlib/v1.8/Distributed/test/distributed_exec.jl:284
   ```
- Doctests: unrelated OOM on the Nanosoldier machine

DilumAluthge commented:
CI is all green now. @vtjnash Are you good with merging this?

nalimilan commented:
Thanks @ararslan! I you feel like upgrading to 1.6 now... :-p

mkitti commented:
Note the ABI breakage when moving to 1.6:
https://abi-laboratory.pro/index.php?view=compat_report&l=libunwind&v1=1.5.0&v2=1.6.0&obj=8ee57&kind=abi

> Field load_offset has been added at the middle position of [unw_dyn_info]
